src/Data/HashMap/Linear.hs

```diff
@@ -100,11 +100,13 @@ data RobinQuery k where
 -- # Construction and Modification
 --------------------------------------------------
 
+
+
 -- | Run a computation using a singleton hashmap
 singleton :: Keyed k =>
-  (k, v) -> (HashMap k v #-> Unrestricted b) -> Unrestricted b
+  (k, v) -> (HashMap k v #-> Unrestricted b) -> b
```

Comment:
That's a bad idea. You may want to have a `singleton'` or what have you, which returns the non-unrestricted version. But you need `Unrestricted` in all generality, otherwise, you wouldn't be able to create a hashmap in the scope of another hashmap.

Reply:
Yes, I see.

Reply:
Or maybe it should be the opposite, and `singleton` should be the `b` one, and `singleton'` the `Unrestricted b` one. Good bikeshedding problem.

My point is that we should not take away the more general version from the API.

Reply:
Not sure which one should be which. Just defaulted to what was there before for now.